src/Ext/Techno/Hooks.Pips.cpp

```diff
@@ -1,8 +1,39 @@
 #include <AircraftTypeClass.h>
 #include <SpawnManagerClass.h>
 #include <TiberiumClass.h>
+#include <TacticalClass.h>
 #include "Body.h"
 
+DEFINE_HOOK(0x6D9070, TacticalClass_RenderLayers_DrawBefore_Foot, 0x6)
+{
+	GET(TechnoClass*, pTechno, ESI);
+	LEA_STACK(Point2D*, pLocation, STACK_OFFSET(0x100, -0xE8));
+	const auto pTypeExt = TechnoTypeExt::ExtMap.Find(pTechno->GetTechnoType());
+
+	if (!pTypeExt->HealthBar_Hide)
+	{
+		if (pTechno->IsSelected && Phobos::Config::EnableSelectBox && !pTypeExt->HideSelectBox)
+			TechnoExt::DrawSelectBox(pTechno, pLocation, &DSurface::ViewBounds, true);
+	}
+
+	return 0;
+}
+
+DEFINE_HOOK(0x6D912B, TacticalClass_RenderLayers_DrawBefore_Building, 0x9)
+{
+	GET(BuildingClass*, pBuilding, ESI);
+	LEA_STACK(Point2D*, pLocation, STACK_OFFSET(0x100, -0x6C));
+	const auto pTypeExt = TechnoTypeExt::ExtMap.Find(pBuilding->Type);
+
+	if (!pTypeExt->HealthBar_Hide)
+	{
+		if (pBuilding->IsSelected && Phobos::Config::EnableSelectBox && !pTypeExt->HideSelectBox)
+			TechnoExt::DrawSelectBox(pBuilding, pLocation, &DSurface::ViewBounds, true);
+	}
+
+	return 0;
+}
```

Comment:
I have come up with another solution.
Additionally, advancing `IsSelected` judgment should be able to filter out most of them first.
```suggestion
DEFINE_HOOK_AGAIN(0x6D9134, TacticalClass_RenderLayers_DrawBefore, 0x5)// BuildingClass
DEFINE_HOOK(0x6D9076, TacticalClass_RenderLayers_DrawBefore, 0x5)// FootClass
{
	GET(TechnoClass*, pTechno, ESI);
	GET(Point2D*, pLocation, EAX);

	if (pTechno->IsSelected && Phobos::Config::EnableSelectBox)
	{
		const auto pTypeExt = TechnoTypeExt::ExtMap.Find(pTechno->GetTechnoType());

		if (!pTypeExt->HealthBar_Hide && !pTypeExt->HideSelectBox)
			TechnoExt::DrawSelectBox(pTechno, pLocation, &DSurface::ViewBounds, true);
	}

	return 0;
}
```